sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySinkMetrics.java

```diff
@@ -326,6 +337,83 @@ public static void reportFailedRPCMetrics(
     updateRpcLatencyMetric(c, method);
   }
 
+  /**
+   * Results of a batch of InsertAll RPCs. Used to update metrics with the {@code
+   * updateStreamingInsertsMetrics} method. Member variables should be thread-safe.
+   */
+  public static class StreamingInsertsResults {
```

Comment:
if this is a data class, can you use AutoValue? 
https://g3doc.corp.google.com/third_party/java_src/auto/value/g3doc/index.md?cl=head

can also annotate with `@ThreadSafe`

also the members are threadsafe individually, but if you need atomic operations across multiple members that will still require a lock (`synchronized (someObject)`)

Reply:
Added a more nuanced comment about the thread safety guarantees of this class.